cmd/alertmanager/main.go

```diff
@@ -476,3 +478,28 @@ func md5HashAsMetricValue(data []byte) float64 {
 	copy(bytes, smallSum)
 	return float64(binary.LittleEndian.Uint64(bytes))
 }
+
+func setAlertStatus(inhibitor *inhibit.Inhibitor, marker types.Marker, silences *silence.Silences) func(model.LabelSet) error {
```

Comment:
Does this need to be inside `main.go`? I feel like it would be better placed close to `types.Marker`, given that `marker.Status` represents the `getAlertStatus` call. What do you think? Might be included in the TODO below.

> ` 		// TODO(beorn7): The following code is almost exactly replicated in notify/notify.go.`

Reply:
The main reason I put this here is because it seemed like a bundling of imperative things specific to this use case, and then passing them on down to the api. I sort of view `marker.Status` being `getAlertStatus` as a coincidence, and that we created the `getAlertStatus` type as a way to abstract away which specific impl we're using.

Since this particular function is tying together three different packages (`silences`, `inhibit`, and `types`), it felt more natural to me to happen in main. My personal (and debatable, of course) philosophy is that it's easier to define interfaces (easier to unit test), and then assemble all of them together at the top level (for a full acceptance test).

This was original inspired by a [destroy all software screencast](https://www.destroyallsoftware.com/screencasts/catalog/functional-core-imperative-shell), but I've probably diverged from the author's original intent. Happy to talk more, but maybe not in a PR :)

The short answer is: I'm happy to move it if you feel strongly about it. Binding several different packages together didn't leave me with a clear signal where to put it, so `main` felt fine for now. When we figure out how to deduplicate this code this should be easy to remove/replace.